tests/unit/LogServiceInitTests.cpp

```diff
@@ -0,0 +1,230 @@
+#include <spdlog/spdlog.h>
+
+#include <cstddef>
+#include <cstdint>
+#include <limits>
+#include <memory>
+#include <ostream>
+#include <string>
+#include <string_view>
+#include <vector>
+using namespace util;
+
+using util::config::Array;
+using util::config::ConfigFileJson;
+using util::config::ConfigType;
+using util::config::ConfigValue;
+
+struct LogServiceInitTests : virtual public ::testing::Test {
+protected:
+    util::config::ClioConfigDefinition config_{
+        {"log_channels.[].channel", Array{ConfigValue{ConfigType::String}}},
+        {"log_channels.[].log_level", Array{ConfigValue{ConfigType::String}}},
+
+        {"log_level", ConfigValue{ConfigType::String}.defaultValue("info")},
+
+        {"spdlog_format", ConfigValue{ConfigType::String}.defaultValue(R"(%Y-%m-%d %H:%M:%S.%f %^%3!l:%n%$ - %v)")},
+        {"spdlog_async", ConfigValue{ConfigType::Boolean}.defaultValue(false)},
+
+        {"log_to_console", ConfigValue{ConfigType::Boolean}.defaultValue(false)},
+
+        {"log_directory", ConfigValue{ConfigType::String}.optional()},
+
+        {"log_rotation_size",
+         ConfigValue{ConfigType::Integer}.defaultValue(2048).withConstraint(config::gValidateUint32)},
+
+        {"log_directory_max_files",
+         ConfigValue{ConfigType::Integer}.defaultValue(25).withConstraint(config::gValidateUint32)},
+
+        {"log_tag_style", ConfigValue{ConfigType::String}.defaultValue("none")},
+    };
+
+    std::string
+    getLoggerString()
+    {
+        return buffer_.getStrAndReset();
+    }
+
+    void
+    replaceSinks()
+    {
+        auto ostreamSink = std::make_shared<spdlog::sinks::ostream_sink_mt>(stream_);
+
+        for (auto const& channel : Logger::kCHANNELS) {
+            auto logger = spdlog::get(channel);
+            ASSERT_TRUE(logger != nullptr);
+            // It is expected that only stderrSink is present
+            ASSERT_EQ(logger->sinks().size(), 1);
+            logger->sinks().clear();
+            logger->sinks().push_back(ostreamSink);
+        }
+
+        spdlog::set_pattern("%n:%L %v");
+    }
+
+private:
+    StringBuffer buffer_;
+    std::ostream stream_ = std::ostream{&buffer_};
+};
+
+TEST_F(LogServiceInitTests, DefaultLogLevel)
+{
+    auto const parsingErrors = config_.parse(ConfigFileJson{boost::json::object{{"log_level", "warn"}}});
+    ASSERT_FALSE(parsingErrors.has_value());
+    std::string const logString = "some log";
+
+    EXPECT_TRUE(LogService::init(config_));
+    replaceSinks();
+
+    for (auto const& channel : Logger::kCHANNELS) {
+        Logger const log{channel};
+        log.trace() << logString;
+        ASSERT_TRUE(getLoggerString().empty());
+
+        log.debug() << logString;
+        ASSERT_TRUE(getLoggerString().empty());
+
+        log.info() << logString;
+        ASSERT_TRUE(getLoggerString().empty());
+
+        log.warn() << logString;
+        ASSERT_EQ(fmt::format("{}:W {}\n", channel, logString), getLoggerString());
```

Comment:
This still works because the pattern set above, right? but this is not the same as our new default pattern so maybe good to update to match

Reply:
Updated the pattern, but without timestamp